docs-search/src/Docs/Search/JsonCodec.purs

```diff
@@ -0,0 +1,51 @@
+-- | Utilities to interop between different codec libraries
+module Docs.Search.JsonCodec
+  ( fromGeneric
+  , fromJsonUnidirectional
+  , inject
+  ) where
+
+import Prelude
+import Prim.Row (class Cons)
+
+import Codec.Json.Unidirectional.Value (DecodeError)
+import Data.Argonaut.Core (Json, fromString, stringify)
+import Data.Argonaut.Decode as Generic
+import Data.Argonaut.Decode.Error as Generic
+import Data.Argonaut.Encode as Generic
+import Data.Codec.Argonaut (JsonCodec, JsonDecodeError(..))
+import Data.Codec.Argonaut.Common as CA
+import Data.Codec.Argonaut.Sum as CAS
+import Data.Codec.Argonaut.Record as CAR
+import Data.Either (Either(..))
+import Data.Profunctor.Choice (left)
+import Data.Symbol (class IsSymbol)
+import Data.Variant
+import Type.Proxy (Proxy(..))
+
+-- | Equivalent to `Data.Variant.inj`, just uses a Visible Type Application instead of Proxy
+-- | Useful for deriving codecs for sum types
+inject :: forall @sym a r1 r2. Cons sym a r1 r2 => IsSymbol sym => a -> Variant r2
+inject = inj (Proxy :: _ sym)
+
+fromJsonUnidirectional
+  :: forall a
+   . (Json -> Either DecodeError a)
+  -> (Json -> Either JsonDecodeError a)
+fromJsonUnidirectional = map $ left convertError
+  where
+  convertError :: DecodeError -> JsonDecodeError
+  convertError = const $ UnexpectedValue $ fromString ""
```

Comment:
This block of code is why the PR is still in draft mode. I need to figure how to best map error types between Jordan's `json-codecs` and `codec-argonaut`.

Reply:
Print my lib's error and then encode that as JSON?